ejb3/src/main/resources/schema/jboss-ejb-timer-service_3_0.xsd

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+~ JBoss, Home of Professional Open Source
+~ Copyright 2022, Red Hat, Inc., and individual contributors as indicated
+~ by the @authors tag.
+~
+~ Licensed under the Apache License, Version 2.0 (the "License");
+~ you may not use this file except in compliance with the License.
+~ You may obtain a copy of the License at
+~
+~ http://www.apache.org/licenses/LICENSE-2.0
+~
+~ Unless required by applicable law or agreed to in writing, software
+~ distributed under the License is distributed on an "AS IS" BASIS,
+~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+~ See the License for the specific language governing permissions and
+~ limitations under the License.
+-->
+<xs:schema xmlns="urn:timer-service:3.0"
+           targetNamespace="urn:timer-service:3.0"
+           version="3.0"
+           xmlns:jakartaee="https://jakarta.ee/xml/ns/jakartaee"
+           xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           attributeFormDefault="unqualified"
+           elementFormDefault="qualified"
+           xsi:schemaLocation="https://jakarta.ee/xml/ns/jakartaee https://www.jboss.org/schema/jbossas/jboss-ejb3-spec-4_0.xsd">
+    <xs:import namespace="https://jakarta.ee/xml/ns/jakartaee" schemaLocation="https://www.jboss.org/schema/jbossas/jboss-ejb3-spec-4_0.xsd"/>
+
+    <xs:element name="timer-service" substitutionGroup="jakartaee:assembly-descriptor-entry" type="timer-serviceType"/>
+
+    <xs:complexType name="timer-serviceType">
+        <xs:complexContent>
+            <xs:extension base="jakartaee:jboss-assembly-descriptor-bean-entryType">
+                <xs:choice minOccurs="0">
+                    <xs:element name="persistence-store-name" type="xs:string">
+                        <xs:annotation>
+                            <xs:documentation>Identifies the name of the data store to use for persistent timers.</xs:documentation>
+                        </xs:annotation>
+                    </xs:element>
+                    <xs:element name="persistent-timer-management" type="xs:string">
+                        <xs:annotation>
+                            <xs:documentation>Identifies the timer-management provider to use for persistent timers.</xs:documentation>
+                        </xs:annotation>
+                    </xs:element>
+                    <xs:element name="transient-timer-management" type="xs:string">
+                        <xs:annotation>
+                            <xs:documentation>Identifies the timer-management provider to use for persistent timers.</xs:documentation>
+                        </xs:annotation>
+                    </xs:element>
+                </xs:choice>
```

Comment:
Compared to previous version (https://github.com/wildfly/wildfly/blob/main/ejb3/src/main/resources/schema/jboss-ejb-timer-service_2_0.xsd), the structure of this element is different.

In existing version, there is a:
all
    choice (config for persistent timer)
        option 1 for persistent timer
        option 2 for persistent timer
    choice (config for non-persistent timer)

In the new version, it's flattened to a choice element. Any reason for this change?

Reply:
Also looks like the parser for this xml is not updated (TimerServiceMetaDataParser and TimerServiceMetaDataSchema)

Reply:
the xs:all can't have a xs:choice as child

Reply:
@pferraro fyi

Reply:
Yeah, that schema is messed up.  It should look something like:
```xml
    <xs:complexType name="timer-serviceType">
        <xs:complexContent>
            <xs:extension base="javaee:jboss-assembly-descriptor-bean-entryType">
                <xs:choice>
                    <xs:element name="persistence-store-name" type="xs:string">
                        <xs:annotation>
                            <xs:documentation>Identifies the name of the data store to use for persistent timers.</xs:documentation>
                        </xs:annotation>
                    </xs:element>
                    <xs:sequence>
                        <xs:element name="persistent-timer-management" type="xs:string">
                            <xs:annotation>
                                <xs:documentation>Identifies the timer-management provider to use for persistent timers.</xs:documentation>
                            </xs:annotation>
                        </xs:element>
                        <xs:element name="transient-timer-management" type="xs:string">
                            <xs:annotation>
                                <xs:documentation>Identifies the timer-management provider to use for persistent timers.</xs:documentation>
                            </xs:annotation>
                        </xs:element>
                    </xs:sequence>
                </xs:choice>
            </xs:extension>
        </xs:complexContent>
    </xs:complexType>
```
The general idea is that a user would specify either persistence-store-name or persistent-timer-management, but not both.

Reply:
@ehsavoie Why are we adding a new schema version?  Version 2.0 of this schema is already new since WF27.

Reply:
I'll address this separately via https://issues.redhat.com/browse/WFLY-17049